src/Libraries/UnitsUI/ViewModelCustomization.cs

```diff
@@ -0,0 +1,314 @@
+            nodeModel = nodeView.ViewModel.NodeModel;
+            model.PropertyChanged += model_PropertyChanged;
+            ToggleButtonClick = new DelegateCommand(OnToggleButtonClick, CanToggleButton);
+        }
+
+        private void model_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            switch (e.PropertyName)
+            {
+                case "SelectedQuantityConversionSource":
+                    RaisePropertyChanged("SelectedQuantityConversionSource");
+                    break;
+                case "SelectedQuantityConversion":
+                    RaisePropertyChanged("SelectedQuantityConversion");
+                    break;
+                case "SelectedFromConversionSource":
+                    RaisePropertyChanged("SelectedFromConversionSource");
+                    break;
+                case "SelectedToConversionSource":
+                    RaisePropertyChanged("SelectedToConversionSource");
+                    break;
+                case "SelectedFromConversion":
+                    RaisePropertyChanged("SelectedFromConversion");
+                    break;
+                case "SelectedToConversion":
+                    RaisePropertyChanged("SelectedToConversion");
+                    break;
+                case "IsSelectionFromBoxEnabled":
+                    RaisePropertyChanged("IsSelectionFromBoxEnabled");
+                    break;
+                case "SelectionFromBoxToolTip":
+                    RaisePropertyChanged("SelectionFromBoxToolTip");
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Called when Toggle button is clicked.
+        /// Switches the combo box values
+        /// </summary>
+        /// <param name="sender">The sender.</param>
+        /// <param name="e">The <see cref="RoutedEventArgs"/> instance containing the event data.</param>
+        private void OnToggleButtonClick(object obj)
+        {
+            //var undoRecorder = nodeViewModel.WorkspaceViewModel.Model.UndoRecorder;
+            //WorkspaceModel.RecordModelForModification(nodeModel, undoRecorder);
+            dynamoConvertModel.ToggleDropdownValues();
+            nodeViewModel.WorkspaceViewModel.HasUnsavedChanges = true;
+        }
+
+        private bool CanToggleButton(object obj)
+        {
+            return true;
+        }
+    }
+    public class UnitValueOutputDropdownViewModel : NotificationObject
+    {
+        private readonly UnitValueOutputDropdown unitValueDropdownModel;
+       // public DelegateCommand ToggleButtonClick { get; set; }
+        private readonly NodeViewModel nodeViewModel;
+        private readonly NodeModel nodeModel;
+
+       
+        public string DisplayValue
+        {
+            get { return unitValueDropdownModel.DisplayValue; }
+            set
+            {
+                RaisePropertyChanged(nameof(DisplayValue));
```

Comment:
does this not need to set anything?

Reply:
I believe we don't need to set anything here. `unitValueDropdownModel.DisplayValue` gets set whenever the 'CachedValue' property of the model is updated (line 288 of this same file). We never need to `set` the `DisplayValue` in that viewmodel.

Reply:
Right, so is there actually any need for a setter here? could we not just do:
`public string DisplayValue => unitValueDropdownModel.DisplayValue;`

and raise property changed when we set the models display value?